src/bosh-director/lib/bosh/director/deployment_deleter.rb

```diff
@@ -57,5 +59,52 @@ def delete(deployment_model, instance_deleter, vm_deleter)
         deployment_model.destroy
       end
     end
+
+    private
+
+    def unused_subnets(new_subnets, old_subnets)
+      old_subnets.select do |subnet|
+        new_subnets.find { |e| e['name'] == subnet.name }.nil?
+      end
+    end
+
+    def latest_cloud_config(deployment)
+      Models::Config.latest_set_for_teams('cloud', *deployment.teams).max(&:id).raw_manifest
+    end
+
+    def used_cloud_config_state(deployment)
+      latest = Models::Config.latest_set_for_teams('cloud', *deployment.teams).map(&:id).sort
+      if deployment.cloud_configs.empty?
+        'none'
+      elsif deployment.cloud_configs.map(&:id).sort == latest
+        'latest'
+      else
+        'outdated'
+      end
+    end
+
+    def delete_subnet(cloud_factory, subnet)
+      cpi = cloud_factory.get(subnet.cpi)
+      begin
+        @logger.info("deleting unused subnet #{subnet.name}")
+        cpi.delete_network(subnet.cid)
+      rescue StandardError => e
+        @logger.error("failed to delete subnet #{subnet.name}: #{e.message}")
```

Comment:
It seems unusual to attempt to remove the subnet, and then if it fails, remove it from the bosh database, and never retry deleting it.  

Maybe it shouldn't destroy it from the bosh database in the failure state, so that the next time it does a cleanup, bosh can try again.

Reply:
Commented below.